### DISCUSSION TITLE
Does it work well with M2.4.4?

### DISCUSSION THREAD

**Discussion by asannikov** [rafaelstz/deployer-magento2#60] [Ideas]
Since M2.4.4 works under php 8.1, my deployment configuration produces many warnings:

`PHP Deprecated:  Return type of Pimple\Container::offsetExists($id) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///usr/local/bin/dep/vendor/pimple/pimple/src/Pimple/Container.php on line 133`

`Deprecated:  Return type of Deployer\Collection\Collection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///usr/local/bin/dep/src/Collection/Collection.php on line 58`

`PHP Deprecated:  Return type of Symfony\Component\Process\Process::getIterator(int $flags = 0) should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///usr/local/bin/dep/vendor/symfony/process/Process.php on line 605`

Currently, deployer uses version 6.x but there is available 7.x already, which has a pretty different syntax. 
So the question is there any plans to move to 7.x?

**Comment by rafaelstz** [comment]
Hi @asannikov, thank you for reporting it, I'm going to push an update to Deployer 7 and PHP 8.1 soon.

**Comment by avestique** [comment]
hi @rafaelstz, any update here?
